partitions.go

```diff
@@ -221,17 +247,21 @@ func (p *partitions) partitionZKNode(partition int) string {
 }
 
 // getPeers returns the list of peers who have the given partition available.
```

Comment:
You need to update the docstring. However, I still think it'd be cleaner to just return one list, and consider the peers "ordered" by priority. That gives us the freedom to actually add in priority later, if we want to do some light downweighting of peers from which we've seen errors.